ibm/service/kubernetes/resource_ibm_container_ingress_secret_opaque.go

```diff
@@ -73,6 +76,16 @@ func ResourceIBMContainerIngressSecretOpaque() *schema.Resource {
 				Computed:    true,
 				Description: "Status of the secret",
 			},
+			"update_secret": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Updates secret from secrets manager if value is changed (increment each usage)",
+			},
```

Comment:
`update_secret` is never set in the code with `d.Set`. if I get it right, terraform will detect this field as changed on every apply, if the field is not empty.
this can be verified with a test, where you apply the same testAccCheckIBMContainerIngressSecretOpaqueForceUpdate configuration twice.

you can to use `d.Set` and update the state after the `ingressAPI.UpdateIngressSecret `.

( or if this behaviour works for you, the field can be modified to a boolean. )

Reply:
I was unable to replicate the behavior you mentioned either by running the test in the code with the same config again or by manually using my own terraform plan/resources. I see the `update_secret` field reflected in the state file and terraform doesn't execute an update on a repeat apply of the same config should the variable remain unchanged. I also took a look at the above mentioned field `retry_patch_version` whose pattern `update_secret` was modeled after and I didn't see any instances of that implementation using the `d.set` either. Although even in the case it did update every time that would be acceptable behavior. Please let me know if there are any other concerns.

I also went ahead and updated the tests to validate this behavior:
```
make testacc TEST=./ibm/service/kubernetes TESTARGS='-run=TestAccIBMContainerIngressSecretOpaque_ForceUpdate'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./ibm/service/kubernetes -v -run=TestAccIBMContainerIngressSecretOpaque_ForceUpdate -timeout 700m
=== RUN   TestAccIBMContainerIngressSecretOpaque_ForceUpdate
--- PASS: TestAccIBMContainerIngressSecretOpaque_ForceUpdate (84.83s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/kubernetes	86.414s
```

```
make testacc TEST=./ibm/service/kubernetes TESTARGS='-run=TestAccIBMContainerIngressSecretTLS_BasicForceUpdate'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./ibm/service/kubernetes -v -run=TestAccIBMContainerIngressSecretTLS_BasicForceUpdate -timeout 700m
=== RUN   TestAccIBMContainerIngressSecretTLS_BasicForceUpdate
--- PASS: TestAccIBMContainerIngressSecretTLS_BasicForceUpdate (63.88s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/kubernetes	65.472s
```

Reply:
I wasn't sure about this one.
thank you Jared for looking into it!